proto/aserto/directory/model/v3/model.proto

```diff
@@ -6,44 +6,114 @@ option go_package = "github.com/aserto-dev/go-directory/aserto/directory/model/v
 
 import "google/api/field_behavior.proto";
 import "google/protobuf/empty.proto";
-
 import "aserto/directory/common/v3/common.proto";
 
 service Model {
     rpc GetManifest(GetManifestRequest) returns (stream GetManifestResponse) {};
     rpc SetManifest(stream SetManifestRequest) returns (SetManifestResponse) {};
+    rpc DeleteManifest(DeleteManifestRequest) returns (DeleteManifestResponse) {};
+    rpc ListManifests(ListManifestsRequest) returns (ListManifestsResponse) {};
+
+    rpc GetObjectType(GetObjectTypeRequest) returns (GetObjectTypeResponse) {};
+    rpc GetObjectTypes(GetObjectTypesRequest) returns (GetObjectTypesResponse) {};
 }
 
-message GetManifestRequest {
+message Metadata {
     // manifest name (unique, lc-string)
-    string name                                                 = 1     [(google.api.field_behavior) = REQUIRED];
+    string  name                                                  = 1      [(google.api.field_behavior) = REQUIRED];
     // manifest version (semver notation)
-    string version                                              = 2     [(google.api.field_behavior) = REQUIRED];
+    string  version                                               = 2      [(google.api.field_behavior) = REQUIRED];
+}
+
+enum OnDelete {
```

Comment:
Let's save this for a later revision. I think we need to start using the service in order to get a solid understanding of the kinds of behaviors we want to support when there are conflicts between a new manifest and the current one.